test/e2e/state.js

```diff
@@ -0,0 +1,141 @@
+  // recordings. If someone wanted to, they could go in and delete the workspace
+  // or recordings in it or anything. While thats not great, it's also not
+  // the end of the world. If someone does that, we can always change
+  // this code to only run in CI in the main repo and have this as a secret.
+  // It's a lot easier to hardcode it for now though.
+  replayApiKey: "rwk_7XPbO5fhz0bkhANYXtN2dkm74wNQCchXf2OxVgAerTQ",
+  DefaultDispatchServer: "wss://dispatch.replay.io",
+
+  usePlaywright: true,
+  ExampleRecordings: fs.existsSync("./test/example-recordings.json")
+    ? JSON.parse(fs.readFileSync("./test/example-recordings.json"))
+    : {},
+
+  patterns: [],
+  stripeIndex: undefined,
+  stripeCount: undefined,
+  dispatchServer: undefined,
+  shouldRecordExamples: false,
+  shouldRecordViewer: false,
+  recordExamplesSeparately: false,
+  testTimeout: 240,
+  skippedTests: process.env.SKIPPED_TESTS,
+  onlyTarget: undefined,
+
+  // Runtime state
+  count: 1,
+  startTime: Date.now(),
+  timings: {},
+  failures: [],
+
+  // CONSTANTS
+
+  // Creating an example returns this constant if the test should be skipped.
+  SkipTest: "skipped",
+
+  // Creating an example returns this constant if the test should be recorded
+  // as part of the viewer process. The test will be recorded and then stopped to
+  // switch to the viewer, as when using replay normally.
+  RecordExampleInViewer: "record-example-in-viewer",
+};
+
+const usage = `
+  Usage: run.js arguments
+  Arguments:
+    --count N: Run tests N times
+    --pattern PAT: Only run tests matching PAT
+    --record-examples: Record the example and save the recordings locally for testing
+    --record-viewer: Record the viewer while the test is running
+    --record-all: Record examples and save the recordings locally, and record the viewer
+    --separate: Record examples in a separate browser instance.
+    --timeout N: Use a timeout of N seconds for tests (default 240).
+    --target TARGET: Only run tests using given TARGET
+    --server ADDRESS: Set server to connect to (default wss://dispatch.replay.io).
+`;
+function processArgs(state, argv) {
+  for (let i = 2; i < argv.length; i++) {
+    const arg = argv[i];
+    switch (arg) {
+      case "--count":
+        state.count = +argv[++i];
+        break;
+      case "--pattern":
+        state.patterns.push(argv[++i]);
+        break;
+      case "--record-examples":
+        state.shouldRecordExamples = true;
+        break;
+      case "--record-viewer":
+        state.shouldRecordViewer = true;
+        break;
+      case "--record-all":
+        state.shouldRecordViewer = true;
+        state.shouldRecordExamples = true;
+        break;
+      case "--separate":
+        state.shouldRecordExamples = true;
+        state.recordExamplesSeparately = true;
+        break;
+      case "--timeout":
+        state.testTimeout = +argv[++i];
+        break;
+      case "--target":
+        state.onlyTarget = argv[++i];
```

Comment:
Total side-note: I would love to get rid of the `onlyTarget` variable at some point and make it so that passing `--target` just changes targets to `[onlyTarget]`.

Reply:
hmm, where is targets defined? I think i isolated this variable to one spot which makes it a bit easier to modify going forward.

Reply:
`targets` comes in from `manifest.js`